File: Array/TEST/Q3.java
import java.io.FileInputStream;
import java.io.PrintStream;
import java.util.*;

public class Q3 {

    // ------------- FUNCTION AREA: START -------------

    static int[] removeDamagedTickets(Integer[] tickets, int t) {

        int n = tickets.length;
        int[] result = new int[n];

        int index = 0;
        int count = 0;

        for (int i = 0; i < n; i++) {

            if (tickets[i] == t) {

                // Keep only even occurrences (2nd, 4th, ...)
                if (count % 2 == 1) {
                    result[index] = tickets[i];
                    index++;
                }

                count++;

            } else {

                result[index] = tickets[i];
                index++;
            }
        }

        // Remaining positions automatically stay 0
        return result;
    }

    // ------------- FUNCTION AREA: END ---------------

    public static void main(String[] args) throws Exception {

        Scanner sc = new Scanner(System.in);

        if (System.getProperty("ONLINE_JUDGE") == null) {
            System.setIn(new FileInputStream("input.txt"));
            System.setOut(new PrintStream("output.txt"));
        }

        // ---------------- START CODE -------------------

        Integer[] tickets = new Integer[8];

        for (int i = 0; i < tickets.length; i++) {
            tickets[i] = sc.nextInt();
        }

        int t = sc.nextInt();

        int[] result = removeDamagedTickets(tickets, t);

        System.out.println(Arrays.toString(result));

        // ---------------- END CODE ---------------------

        sc.close();

    }
}

File: Array/HW/Assignment 1/Q3.py
def removeDamagedTickets(tickets, t):
    occurrence_count = 0
    new_tickets = []
    for val in tickets:
        if val == t:
            if occurrence_count % 2 == 0:
                occurrence_count += 1
                continue
            occurrence_count += 1
        new_tickets.append(val)

    while len(new_tickets) < len(tickets):
        new_tickets.append(0)

    return new_tickets


if __name__ == "__main__":
    tickets = [4, 9, 4, 2, 4, 7, 4, 0]
    t = 4

    print(removeDamagedTickets(tickets, t))
